File: packages/sdk/ios/core/BLEIntSDK/Classes/ClientError.swift
//
//  ClientError
//
//  Copyright © 2022 2hire.io. All rights reserved.
//

import Foundation

public enum ClientError: Error {
    case InvalidData
    case InvalidState
    case InvalidSession(errorCode: String)
    case InvalidCommand(errorCode: String)
}

extension ClientError: LocalizedError {
    public var errorDescription: String? {
        switch self {
        case .InvalidData:
            return NSLocalizedString("Client invalid data received", comment: "")
        case .InvalidState:
            return NSLocalizedString("Client invalid state", comment: "")
        case .InvalidSession(let errorCode):
            return NSLocalizedString("Client invalid session (\(errorCode))", comment: "")
        case .InvalidCommand(let errorCode):
            return NSLocalizedString("Client invalid command (\(errorCode))", comment: "")
        }
    }
}

File: packages/sdk/android/core/src/main/java/io/twohire/bleintsdk/client/ClientError.kt
package io.twohire.bleintsdk.client

import io.twohire.bleintsdk.protocol.WritableTLState
import io.twohire.bleintsdk.utils.ErrorDescription

enum class ClientError : ErrorDescription {
    INVALID_DATA {
        override val description: String
            get() = "Client invalid data received"
    },
    INVALID_STATE {
        override val description: String
            get() = "Client invalid state"
    },
    INVALID_SESSION {
        override val description: String
            get() = "Client invalid session"
    },
    INVALID_COMMAND {
        override val description: String
            get() = "Client invalid command"
    };
}